lib/pages/search/search.dart

```diff
@@ -61,19 +60,127 @@ class SearchController extends State<Search>  with ComparablePresentationContact
+          .any((existingUser) => existingUser.displayName == user.displayName);
+
+        if (!isDuplicateUser) {
+          recentChatListPresentationSearch.add(user);
+        }
+
+        if (recentChatListPresentationSearch.length == 6) {
+          break; // Stop getting participants after 6 or more have been added
+        }
+      }
+
+      if (recentChatListPresentationSearch.length == 6) {
+        break; // Stop getting participants after 7 or more have been added
+      }
+    }
+
+    Logs().d('SearchController::getContactsFromRecentChat() - event: $recentChatListPresentationSearch');
+
+    return recentChatListPresentationSearch;
+  }
+
+  List<PresentationSearch> getContactsAndRecentChatStream(Either<Failure, GetContactAndRecentChatSuccess> event) {
+    return event.fold<List<PresentationSearch>>(
+      (failure) => <PresentationSearch>[],
+      (success) {
+        final contactsAndRecentChat = success.search.toPresentationSearch();
+        updateLastContactAndRecentChatIndex(contactsAndRecentChat.length);
+        return contactsAndRecentChat;
+      },
+    ).toList();
+  }
+
+  void listenSearchContactAndRecentChat() {
+    searchContactAndRecentChatController?.getContactAndRecentChatStream.stream.listen((event) {
+      Logs().d('SearchController::getContactAndRecentChatStream() - event: $event');
+      getContactAndRecentChatStream.add(event);
+    });
+  }
+
+  void goToChatScreen(PresentationSearch presentationSearch) {
+    if (presentationSearch.isContact) {
+      showFutureLoadingDialog(
+        context: context,
+        future: () async {
+          if (presentationSearch.directChatMatrixID != null && presentationSearch.directChatMatrixID!.isNotEmpty) {
+            final roomId = await Matrix.of(context).client.startDirectChat(presentationSearch.directChatMatrixID!);
+            VRouter.of(context).toSegments(['rooms', roomId]);
+          }
+        },
+      );
+    } else {
+      if (presentationSearch.matrixId != null) {
+        VRouter.of(context).toSegments(['rooms', presentationSearch.matrixId!]);
+      }
+    }
+  }
+
+  void goToChatScreenFormRecentChat(User user) async {
+    Logs().d('SearchController::getContactAndRecentChatStream() - event: $user');
+    final roomIdResult = await showFutureLoadingDialog(
+      context: context,
+      future: () => user.startDirectChat(),
+    );
+    if (roomIdResult.error != null) return;
+    VRouter.of(context).toSegments(['rooms', roomIdResult.result!]);
+  }
+
   @override
   void initState() {
+    searchContactAndRecentChatController = SearchContactAndRecentChatController(context);
+    isSearchModeNotifier = ValueNotifier(false);
     listenContactsStartList();
+    listenSearchContactAndRecentChat();
     super.initState();
-    fetchContactsController.fetchCurrentTomContacts(limit: limitRecentContacts);
+    SchedulerBinding.instance.addPostFrameCallback((_) async {
+      if (mounted) {
+        searchContactAndRecentChatController?.init();
```

Comment:
why put it in ScheduleBinding ?

Reply:
```
  void _getContactAndRecentChat() {
    _searchContactsAndRecentChatInteractor.execute(
      rooms: Matrix.of(context).client.rooms,
      matrixLocalizations: MatrixLocals(L10n.of(context)!),
      keyword: '',
      limitRecentChats: limitPrefetchedRecentChats,
      limitContacts: limitSearchingPrefetchedRecentContacts,
    ).listen((event) {
      getContactAndRecentChatStream.add(event);
    });
  }
```

Currently I use `context` when initState so I need to put it in ScheduleBinding to make sure the context is created first.